### ISSUE TITLE
Multiple Computers

### ISSUE THREAD

**Issue by salimhurjuk** [kieranwalsh/PowerShell#3]
How can i scan multiple computers from this script

**Comment by PatchesMcMatt** [comment]
This part of the script enables it to scan multiple computers.


```
$WindowsComputers = (Get-ADComputer -Filter {
    (OperatingSystem  -Like 'Windows*') -and (OperatingSystem -notlike '*Windows 10*')
}).Name|
Sort-Object
```

**Comment by salimhurjuk** [comment]
Do i have to simply run this script or i have provide any informations

**Comment by onlabs-fr** [comment]
You can run this script as it but it will ignore Windows 10.

**Comment by kieranwalsh** [comment]
Yeah, this is for Active Directory domains really. It'll grab all machines from AD that are Windows based, ignoring Windows 10, and then scan all of them.

**Comment by salimhurjuk** [comment]
Thanks @kieranwalsh

**Comment by battousai7788** [comment]
Having some issues with this script where when it runs it list (0) unpatched, (0) patched, (1) off/untested, but no computer listed below off.  Not sure where the script is failing for me.

**Comment by onlabs-fr** [comment]
What is your PowerShell version ? It has a difference between PowerShell v2.0 and above.
In my AD 2008 R2 (v2.0) _(Get-AdUser -Filter *).Name_ returns nothing but it is OK with 2012 (PowerShell v4.0).
I think the correct filter works with all PShell version has to be :
$WindowsComputers = Get-ADComputer -Filter {OperatingSystem -Like 'Windows*'} | Select -ExpandProperty Name | Sort-Object
posted in pull request : [https://github.com/kieranwalsh/PowerShell/pull/9](url)

**Comment by battousai7788** [comment]
I think that was my issue, I went ahead and updated to v4.0.  Thanks for the help!

**Comment by junaidma78** [comment]
The script is good, but how do I filter target OU's , where do I add specific OU paths into the script, please assist.
Thanks you in advance appreciate it.
Regards
Junaid

**Comment by PatchesMcMatt** [comment]
@junaidma78, you'd adjust the Active Directory query.  Something like this, for example,

$WindowsComputers = (Get-ADComputer -Filter {
    (OperatingSystem  -Like 'Windows*') -and (OperatingSystem -notlike '*Windows 10*')
} -SearchBase "CN=Computers,DC=DOMAIN,DC=LOCAL").Name|
Sort-Object

**Comment by junaidma78** [comment]
Thank you PatchesMcMatt will give it a try now

**Comment by junaidma78** [comment]
Thanks McMatt all good

**Comment by jward448** [comment]
Thanks, @kieranwalsh. Your script allowed me to create some scripts that execute a nightly scan that produces some web pages that our staff can check to track our patching progress. Free for anyone who is still trying to make sure they get everything patched here: https://github.com/jward448/wcpatchreporting
